src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -31,14 +31,20 @@ class TemplateFactory implements UI\ITemplateFactory
 	/** @var Nette\Caching\IStorage */
 	private $cacheStorage;
 
+	/** @var string */
+	private $templateClassName;
+
 
 	public function __construct(ILatteFactory $latteFactory, Nette\Http\IRequest $httpRequest = NULL,
-		Nette\Security\User $user = NULL, Nette\Caching\IStorage $cacheStorage = NULL)
+		Nette\Security\User $user = NULL, Nette\Caching\IStorage $cacheStorage = NULL, $templateClassName = Template::class)
 	{
 		$this->latteFactory = $latteFactory;
 		$this->httpRequest = $httpRequest;
 		$this->user = $user;
 		$this->cacheStorage = $cacheStorage;
+
+		\Nette\Utils\Validators::assert($templateClassName, 'string');
```

Comment:
Code would be inconsistent with other strings (scalar) types in methods. 

Rather check if this class implements `ITemplate` interface. Because it would fail later and now it silently pass in here creating invalid object.